### PR TITLE
merging chatfriend changes to repo

### PR DESCRIPTION
**Pull request by mikth3372**

### CODE REVIEW

File: client-react-chatfriend/src/components/ChatInterface.js [hunk 1]
```diff
@@ -61,31 +67,58 @@ function ChatInterface({ session, clearChat, handleCompleteTask }) {
...
+      }
+        else {
         handleTextMessage(event);
       }
     };
 
     ws.current.onerror = (event) => {
       console.error("WebSocket error:", event);
-      enqueueSnackbar("WebSocket error observed", { variant: "error" });
+      // enqueueSnackbar("WebSocket error observed", { variant: "error" });
```
Comment (abhay97ps) [inline client-react-chatfriend/src/components/ChatInterface.js:86]: do not comment out error toasters

File: client-react-chatfriend/src/components/InteractionMode.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { useEffect } from "react";
+import {  fetchData } from "../utils/api";
+
+function InteractionMode({ selectedMode, handleModeChange }) {
+    const onModeChange = (e) => {
+        handleModeChange(e.target.value);
+      };
+
+
+    // useEffect(()=>{
```
Comment (abhay97ps) [inline client-react-chatfriend/src/components/InteractionMode.js:11]: nit: tech debt delete comment if no immediate use

**Review by abhay97ps — DISMISSED** [review]
Great Work|
Help me, make a list of feature you build, then create a Asana ticket for gritcoach. We should be tracking Frontend changes that need to updated as needed.

**Comment by mikth3372** [comment]
Sure I will create the list and add it on the asana ticket.

Also I have fixed the conflict.

**Review by abhay97ps — APPROVED** [review]

**Pull request merged**